client-libraries/rust/modality-ingest-client/src/types.rs

```diff
@@ -265,6 +312,32 @@ impl Encode for AttrVal {
     }
 }
 
+impl FromStr for AttrVal {
+    type Err = std::convert::Infallible;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // N.B. Eventually will have to do more advanced deserialization here
+        // as our AttrVals support more interesting variants.
+        Ok(if let Ok(v) = s.to_lowercase().parse::<bool>() {
+            v.into()
+        } else if let Ok(v) = s.parse::<i64>() {
+            v.into()
+        } else if let Ok(v) = s.parse::<i128>() {
+            v.into()
+        } else if let Ok(v) = s.parse::<f64>() {
+            v.into()
+        } else if let Ok(v) = s.parse::<Nanoseconds>() {
+            v.into()
+        } else if let Ok(v) = s.parse::<LogicalTime>() {
+            v.into()
+        } else if let Ok(v) = s.parse::<Uuid>() {
+            v.into()
+        } else {
+            AttrVal::String(s.trim_matches('"').into())
```

Comment:
Right now, I'm accepting both `foo` and `"foo"` as an `AttrVal::String`. My original assumption was that anything that didn't parse as something else should parse as a string, but I'm not sure if that's really what we want to go with. I'm absolutely of two minds about it, so I'd appreciate a tie-break opinion.

1. Only accept strings that are wrapped in `"`.
2. Accept strings that are wrapped in `"` (good for ensuring things aren't parsed as something else), but also interpret anything else not otherwise parsed as a string.

Reply:
This is a toughy. Thinking...

Reply:
To add a bit more context. This is only for the `FromStr` impl, which is essentially only used for `my_str.parse()`. We also have `From<&str>` & `From<String>` which are independent. So this is only deciding for the case where we're trying to "Parse this unknown `AttrVal` from a string" and not "I have something that is explicitly a string that should `.into()` an `AttrVal::String`.

I think that makes the "require quotes" option viable. I'm just not sure if we like the idea of errororing out while parsing the config file or crossing fingers in ambiguous cases more.

Reply:
I think we want to do option 2 here. Probably ought to trim both `"` and `'`, in truth.